SU2_CFD/include/numerics_simd/flow/convection/common.hpp

```diff
@@ -32,37 +32,129 @@
 #include "../variables.hpp"
 #include "../../../variables/CNSVariable.hpp"
 
+/*!
+ * \brief Blended difference for U-MUSCL reconstruction.
+ * \param[in] grad_proj - Gradient projection at point i: dot(grad_i, vector_ij).
+ * \param[in] delta - Centered difference: V_j - V_i.
+ * \param[in] kappa - Blending parameter.
+ * \return Blended difference for reconstruction from point i.
+ */
+FORCEINLINE Double umusclProjection(Double grad_proj,
+                                    Double delta,
+                                    Double kappa) {
+  /*-------------------------------------------------------------------*/
+  /*--- Reconstruction will be V_L = V_i + 0.25 * dV_ij^kappa.      ---*/
+  /*--- Scale dV_ij^cent by 0.5 here and return 0.5 * dV_ij^kappa   ---*/
+  /*--- for backward compatibility (i.e. when kappa==0 and this     ---*/
+  /*--- function isn't called, the reconstruction will be           ---*/
+  /*--- V_L = V_i + 0.5 * proj, otherwise V_L = V_i + 0.5 * blend). ---*/
+  /*-------------------------------------------------------------------*/
+  const Double cent = 0.5 * delta;
+
+  /*--- Upwind difference: dV_ij^upw = 2 grad(Vi) dot vector_ij - dV_ij^cent ---*/
+  const Double upw = grad_proj - cent;
+
+  /*--- Blended difference: dV_ij^kappa = (1-kappa)dV_ij^upw + (1+kappa)dV_ij^cent ---*/
+  return (1.0 - kappa) * upw + (1.0 + kappa) * cent;
+}
+
+/*!
+ * \brief MUSCL reconstruction of the specified variable.
+ * \param[in] grad_i - Gradient vector at point i.
+ * \param[in] delta - Centered difference: V_j - V_i.
+ * \param[in] vector_ij - Distance vector from i to j.
+ * \param[in] iVar - Variable index.
+ * \param[in] umuscl - If true, use U-MUSCL blended difference.
+ * \param[in] kappa - Blending coefficient.
+ * \param[in] relax - Newton-Krylov relaxation.
+ * \return Variable reconstructed from point i.
+ */
+template<class GradType, size_t nDim>
+FORCEINLINE Double musclReconstruction(const GradType& grad,
+                                       const Double delta,
+                                       const VectorDbl<nDim>& vector_ij,
+                                       size_t iVar,
+                                       bool umuscl,
+                                       Double kappa,
+                                       Double relax) {
+  /*--- Gradient projection ---*/
+  Double proj = dot(grad[iVar], vector_ij);
+
+  if (umuscl) {
+    /*--- Blended difference: dV_ij^kappa = (1-kappa)dV_ij^upw + (1+kappa)dV_ij^cent ---*/
```

Comment:
No need to repeat the formulation.
```suggestion
```

Reply:
For the CSolver (non-SIMD) reconstruction you requested using the same code path regardless of muscl_kappa. Should I also get rid of the `if (umuscl)` for SIMD numerics?